### PR TITLE
application: serial_lte_modem: Miscellaneous minor updates

### PR DESCRIPTION
**Pull request by junqingzou**
Add CONFIG_NRF_CLOUD_IPV6 in prj.conf, default not defined
Correct modem fault URC format
Correct socket poll output (uint16_t) and doc
Remove unused header files

Signed-off-by: Jun Qing Zou <jun.qing.zou@nordicsemi.no>

### CODE REVIEW

File: applications/serial_lte_modem/prj.conf [hunk 1]
```diff
@@ -72,6 +72,7 @@ CONFIG_SETTINGS_FCB=y
 CONFIG_FCB=y
 
 # nRF Cloud
+#CONFIG_NRF_CLOUD_IPV6=y
```
Comment (eivindj-nordic) [inline applications/serial_lte_modem/prj.conf:75]: In the commit message, it says "default not defined". However, here it is defined.

File: applications/serial_lte_modem/src/main.c [hunk 2]
```diff
@@ -76,7 +76,7 @@ static void on_modem_failure_shutdown(struct k_work *work)
 	int ret = nrf_modem_lib_shutdown();
 
 	ARG_UNUSED(work);
-	sprintf(rsp, "#XMODEM: SHUTDOWN,%d", ret);
+	sprintf(rsp, "\r\n#XMODEM: SHUTDOWN,%d\r\n", ret);
```
Comment (eivindj-nordic) [inline applications/serial_lte_modem/src/main.c:79]: Should this be added to the response in `nrf_modem_fault_handler` as well?

**Review by FrancescoSer — APPROVED** [review]

**Review by pirun — APPROVED** [review]

File: applications/serial_lte_modem/prj.conf (hunk 1 above)
Comment (junqingzou) [reply applications/serial_lte_modem/prj.conf:75]: It's commented out by default.
Customer could uncomment out this line for quick test of nRF Cloud in IPv6 network.

File: applications/serial_lte_modem/src/main.c (hunk 2 above)
Comment (junqingzou) [reply applications/serial_lte_modem/src/main.c:79]: Yes, you're quite right.

**Review by rlubos — APPROVED** [review]

**Review by eivindj-nordic — APPROVED** [review]

**Pull request merged**